performance_test/src/utilities/cpu_usage_tracker.hpp

```diff
@@ -0,0 +1,242 @@
+// Copyright 2017 Apex.AI, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef UTILITIES__CPU_USAGE_TRACKER_HPP_
+#define UTILITIES__CPU_USAGE_TRACKER_HPP_
+
+
+#include <sys/times.h>
+#include <sys/types.h>
+
+#include <unistd.h>
+
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace performance_test
+{
+
+struct CpuInfo
+{
+  uint32_t m_cpu_cores = 0;
```

Comment:
Make this a proper class with constructor and const members.

Reply:
Done